Python/pylifecycle.c

```diff
@@ -2403,6 +2405,7 @@ Py_EndInterpreter(PyThreadState *tstate)
        when they attempt to take the GIL (ex: PyEval_RestoreThread()). */
     _PyInterpreterState_SetFinalizing(interp, tstate);
 
+    daemon_threads_exited(tstate);
```

Comment:
I don't think this does anything here. If I understand correctly, we don't support daemon threads with multiple interpreters. We have a `Py_FatalError("not the last thread")` check above that ensures that the current thread is the only active thread left.